.github/workflows/build-and-tag-maven.yml

```diff
@@ -49,6 +49,7 @@ jobs:
           echo "VERSION=$VERSION" >> $GITHUB_ENV
 
       - name: Login to DockerHub Registry
+        if: ${{ github.event_name == 'push' }}
```

Comment:
additionally, which is the objective here? I am not getting it, please add some context 🙏🏻

Reply:
The objective is to avoid having to log in on DockerHub on PRs, especially dependabot ones because it would require to add corresponding secrets to the dependabot secrets.
I did explain a lot of this on the jira issue, but I can add more comments here if that's better.

Reply:
I read the issue but not the comments, sorry for that 🤐 now it's clear!

Reply:
> "Require branches to be up to date before merging" is not really helping

true, imagine if a bunch of dependabot PR get opened in the same batch, you are going to rebase/merge the remaining PRs after each automerge (and spend a lot of build minutes for that)

and unluckily dependabot doesn't support [updating multiple dependencies in a single PR](https://github.com/dependabot/dependabot-core/issues/1190#issuecomment-1071367790)

at this point I would suggest disabling that option if is generating more headaches than value and/or try to investigate some other tools (recently I am looking at [updatecli](https://www.updatecli.io/)

Reply:
Thanks for suggesting so, that would certainly help.
Now to test the behavior I would need that PR to be merged first. Are you ok keeping that check as is (again, the same condition is used multiple times in this workflow already)?